processing/src/main/java/io/druid/query/ordering/StringComparators.java

```diff
@@ -293,12 +297,57 @@ public String toString()
     }
   }
 
+  public static class StrlenComparator implements StringComparator
+  {
+    @Override
+    public int compare(String s, String s2)
+    {
+      if(s == s2) {
+        return 0;
+      }
+      // null first
+      if (s == null) {
+        return -1;
+      }
+      if (s2 == null) {
+        return 1;
+      }
+      
+      int res = Ints.compare(s.length(), s2.length());
+      if (res == 0) {
+        res = s.compareTo(s2);
+      }
+      return res;
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+
+      return true;
+    }
+
+    @Override
+    public String toString()
+    {
+      return StringComparators.STRLEN_NAME;
+    }
+  }
+
   public static StringComparator makeComparator(String type)
```

Comment:
This method is used by OrderByColumnSpec for groupby queries, making instances from strings of a Map object, as in.
https://github.com/druid-io/druid/blob/master/processing/src/main/java/io/druid/query/groupby/orderby/OrderByColumnSpec.java#L72

IMHO, this appears to be changed beforehand.